cargo-flash/src/main.rs

```diff
@@ -124,21 +124,30 @@ fn main_try() -> Result<(), failure::Error> {
         args.remove(index);
     }
 
-    Command::new("cargo")
+    let status = Command::new("cargo")
         .arg("build")
         .args(args)
         .stdout(Stdio::inherit())
         .stderr(Stdio::inherit())
         .spawn()?
         .wait()?;
 
+    if !status.success() {
+        std::process::exit(1);
+    }
+
     println!("    {} {}", "Flashing".green().bold(), path_str);
 
     let device = {
         let mut list = daplink::tools::list_daplink_devices();
         list.extend(stlink::tools::list_stlink_devices());
 
-        list.remove(0)
+        if !list.is_empty() {
+            list.remove(0)
+        } else {
+            eprintln!("    {} No supported probe was found.", "Error".red().bold());
+            std::process::exit(1);
+        }
```

Comment:
This could probably be something like

```rust
list.pop().unwrap_or_else(|| {
    eprintln...
})
```

which I find to be a bit more readable.

Reply:
I agree; will implement this.